### ISSUE TITLE
Parentless Projectiles causing mass havoc

### ISSUE THREAD

**Issue by GiggleHub** [CapsAdmin/pac3#444]
I must ask the visual aspect of projectiles be removed until they have become properly functional
The major bug leading me to ask this is when the visual part of a projectile becomes 'parentless' and begins parenting itself to the closest appropriate entity. This can include players, props, entities, npcs, etc. etc.
This has ranged from very minor cases where players complain a can is unwillingly attached to them
To extreme cases where entire contraptions could be missing portions of it, in its place beings some sort of energy ball or what have you
Example :
![](https://cdn.discordapp.com/attachments/263801829299716106/334128512929890304/unknown.png)
Here you can see a prop I spawned (and a few others in the corner)
It has an obnoxiously huge trail as the result of a parentless projectile attaching itself to it. During the time of the event, players werecomplaining rampantly about the red trail attaching to themselves, and while I've taken efforts to try and identify who was at fault, even ignoring costumes piece by piece, these projectiles persist. This makes fixing the problem for the entire server much more difficult

In the case of a metastruct (and maybe the official PAC3 server) "fullupdate" is an easy enough solution. However on other servers, explaining to a heavily confused player to type "record 0" then "stop" into console is a bit more of an issue. It can be especially tedious if someone is going ham with projectiles
making it a necessity to use these commands in a mere matter of minutes. And despite all of this, some projectiles completely remove rendering from the original prop forcing people to delete the prop, and place it again. This can become extremely irritating when much larger constructs are being pasted, built, or otherwise worked on

As if this wasn't bad enough, there was one such case where (you guessed it) a Sans PAC3 Costume was use to make projectiles that launched more projectiles. So these parentless projectiles proceeded to spew out even more projectiles, bumping players around from random locations on the map and even killing some from over half-way across the map! These projectiles then proceeded to become parentless themselves, exponentially worsening the problem.

Referring back to the larger constructs, an example includes one scenario when people were building cities on a server and some person's projectiles were so obnoxious that entire chunks of skyscrapers were missing. As if a horrible recreation of a post-apocalyptic city-scape wasn't enough, I could barely scrounge enough frames to justify playing on the server.

One person chose to defend themselves by saying that they remain tame with their projectiles, only making small bullets over flashy effects. This 'tiny bullet' defense is shot down once it parents itself to any prop slightly larger than the bullet itself. This can potentially result in completely hiding the object, leaving only the bullet at the origin of the now hidden model.

tl;dr : PAC3 Projectiles shouldn't have a visual component (better seen as the "outfit part name" field for the projectile part) to them until they stop de-parenting from the projectile entity itself

Update:

Another example for reference, this is before using 'fullupdate'
![image](https://user-images.githubusercontent.com/8467172/28741789-5db668e4-73ec-11e7-936d-57550afb4b11.png)
And this is the result after using fullupdate, notice the props themselves are still missing.
![image](https://user-images.githubusercontent.com/8467172/28741790-72bb15dc-73ec-11e7-90fa-004614025b9c.png)

Here's a player example, courtesy of "Mad Matrix"
![image](https://user-images.githubusercontent.com/8467172/28757805-ec46170e-7558-11e7-941e-cad68c4b77fe.png)
 
This is just insult to injury:
![image](https://user-images.githubusercontent.com/8467172/28742047-5e1ef344-73f3-11e7-8492-75fe45268748.png)

Update:

Rather than flooding this post with more pictures, I'm just going to link an album containing every single case of projectiles malfunctioning from here on out.
http://imgur.com/a/CDptR

**Comment by ToastyEngineer** [comment]
I believe a good portion of people use projectiles visually. So said problem can't be fixed by removing the visual part of it. 
However I do agree with you, but instead of removing it, perhaps have it fixed so it doesn't happen anymore in the first place.

**Comment by GiggleHub** [comment]
Only issue with that is there have been attempts made prior to this issue being submitted to fix the bug. However, none of them gave any conclusive results, leaving us where we are now.
People can learn to work around or not rely on projectiles having visual components for the time being until a solid solution is found.

**Comment by CapsAdmin** [comment]
this is fixed now isn't it?
